### PR TITLE
US-6303 Added TimeDilation method to accelerate/slow the passage of time in an app 

### PR DESCRIPTION
**Pull request by simongranger**
This will allow to increase the speed of timeline to reduce test time and reduce timing related issues
Slowing down the app can be useful for debugging

There is a corresponding change in https://github.com/YOU-i-Labs/uswish

### CODE REVIEW

File: lib/commands/settings.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import { errors } from 'appium-base-driver';
 import log from '../logger';
+import { youiEngineDriverReturnValues } from '../utils';
```
Comment (jplandry) [inline lib/commands/settings.js]: Looking at the error codes in `util` I wonder if these are defined somewhere in Appium?

Comment (simongranger) [inline lib/commands/settings.js]: hmmm. Yes, they seem to be defined in errors.js within appium-base-driver

Comment (simongranger) [inline lib/commands/settings.js]: but the enum value doesn't match the ones used by the engine:     /*
     \details
     https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/errorhandler.py
     
     """
     Error codes defined in the WebDriver wire protocol.
     """
     # Keep in sync with org.openqa.selenium.remote.ErrorCodes and errorcodes.h
     SUCCESS = 0
     NO_SUCH_ELEMENT = [7, 'no such element']
     NO_SUCH_FRAME = [8, 'no such frame']
     UNKNOWN_COMMAND = [9, 'unknown command']
     STALE_ELEMENT_REFERENCE = [10, 'stale element reference']
     ELEMENT_NOT_VISIBLE = [11, 'element not visible']
     INVALID_ELEMENT_STATE = [12, 'invalid element state']
     UNKNOWN_ERROR = [13, 'unknown error']
     ELEMENT_IS_NOT_SELECTABLE = [15, 'element not selectable']
     JAVASCRIPT_ERROR = [17, 'javascript error']
     XPATH_LOOKUP_ERROR = [19, 'invalid selector']
     TIMEOUT = [21, 'timeout']
     NO_SUCH_WINDOW = [23, 'no such window']
     INVALID_COOKIE_DOMAIN = [24, 'invalid cookie domain']
     UNABLE_TO_SET_COOKIE = [25, 'unable to set cookie']
     UNEXPECTED_ALERT_OPEN = [26, 'unexpected alert open']
     NO_ALERT_OPEN = [27, 'no such alert']
     SCRIPT_TIMEOUT = [28, 'script timeout']
     INVALID_ELEMENT_COORDINATES = [29, 'invalid element coordinates']
     IME_NOT_AVAILABLE = [30, 'ime not available']
     IME_ENGINE_ACTIVATION_FAILED = [31, 'ime engine activation failed']
     INVALID_SELECTOR = [32, 'invalid selector']
     MOVE_TARGET_OUT_OF_BOUNDS = [34, 'move target out of bounds']
     INVALID_XPATH_SELECTOR = [51, 'invalid selector']
     INVALID_XPATH_SELECTOR_RETURN_TYPER = [52, 'invalid selector']
     METHOD_NOT_ALLOWED = [405, 'unsupported operation']
     */

**Comment by jplandry** [comment]
+1

**Pull request merged**